apis/v1alpha1/opentelemetrycollector_types.go

```diff
@@ -48,9 +63,15 @@ const (
 // SEE: OpenTelemetryCollector.spec.ports[index].
 type Ingress struct {
 	// Type default value is: ""
-	// Supported types are: ingress
+	// Supported types are: ingress, route
 	Type IngressType `json:"type,omitempty"`
 
+	// RuleType defines how Ingress exposes collector receivers.
+	// Either multiple hosts with subdomain per receiver port are created or
+	// a single host with multiple paths.
+	// Default is path.
+	RuleType IngressRuleType `json:"ruleType,omitempty"`
```

Comment:
better naming ideas are appreciated :)

Reply:
What about `exposeMethod` instead of `ruleType`?

Reply:
Actually, it might be better to keep the name aligned with the upstream ingress settings. 

It seems that users don't like that operators re-invent existing concepts https://github.com/open-telemetry/opentelemetry-operator/issues/1987